frontend/public/components/routes/create-route.tsx

```diff
@@ -196,11 +227,54 @@ export class CreateRoute extends React.Component<null, CreateRouteState> {
       }));
   }
 
+  newAltServiceEntry() {
+    return {
+      entry: {
+        service: null,
+        weight: 100,
+      },
+      uid: _.uniqueId(),
+    };
+  }
+
+  addAltServiceEntry(){
+    this.setState({
+      alternateServices: _.concat(this.state.alternateServices, this.newAltServiceEntry()),
+    });
+  }
+
+  removeAltServiceEntry(alternateServiceID){
+    const updatedServiceEntriesArray = [...this.state.alternateServices];
+    updatedServiceEntriesArray.splice(alternateServiceID, 1);
+    this.setState({
+      alternateServices: updatedServiceEntriesArray,
+    });
+  }
+
+  onDataChanged(updatedEntry, entryID) {
+    const updatedServiceEntriesArray = [...this.state.alternateServices];
+    const updatedEntryData = {
+      uid: updatedServiceEntriesArray[entryID].uid,
+      entry: updatedEntry,
+    };
+    updatedServiceEntriesArray[entryID] = updatedEntryData;
+    this.setState({
+      alternateServices: updatedServiceEntriesArray,
+    });
+  }
+
   render() {
     const title = 'Create Route';
-    const { loaded, services, service, portOptions, targetPort, termination } = this.state;
+    const { loaded, services, service, portOptions, targetPort, termination, alternateServices } = this.state;
     const serviceOptions = {};
     _.each(_.sortBy(services, 'metadata.name'), ({ metadata: { name } }) => serviceOptions[name] = <ResourceName kind="Service" name={name} />);
+    const configuredServices = _.isEmpty(service) ? [] : [_.get(service, 'metadata.name')];
+    _.each(alternateServices, serviceEntry => {
+      if (!_.isEmpty(_.get(serviceEntry, 'entry.service'))) {
+        configuredServices.push(_.get(serviceEntry, 'entry.service'));
+      }
+    });
```

Comment:
It's better to use a `Set` instead of an array

```ts
const configuredServices = new Set<string>();
if (service) {
  configuredServices.add(service.metadata.name);
}
_.each(alternateServices, ({service}) => configuredServices.add(service));
```

Then later

```ts
!configuredServices.has(key)
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set